release.py

```diff
@@ -0,0 +1,848 @@
+    'tracing': 'tracing/pyproject.toml',
+    'uvlock': 'uv.lock',
+}
+
+
+def get_latest_release_tag(repo: github.Repository.Repository, branch_name: str) -> str | None:
+    """Get the latest release tag from the repository."""
+    releases = repo.get_releases()
+
+    for release in releases:
+        if release.draft:
+            continue
+
+        if 'maintenance' in branch_name:
+            version = branch_name.split('-')[0]
+            if version not in release.tag_name:
+                continue
+
+        return release.tag_name
+
+    return None
+
+
+def bump_minor_version(version: str) -> str:
+    """Bump minor version."""
+    major, minor, patch = map(int, version.split('.'))
+    return f'{major}.{minor + 1}.{patch}'
+
+
+def get_new_tag_for_release(
+    owner: str, repo: github.Repository.Repository, branch_name: str
+) -> str:
+    """Get a new tag for release.
+
+    Default value is generated by increasing the minor version of the latest release's tag.
+
+    Can be overridden with a user-provided version.
+    """
+    latest_tag = get_latest_release_tag(repo, branch_name)
+
+    suggested_tag = ''
+    if not latest_tag:
+        logger.info(f'No version tag found in branch "{branch_name}"')
+    else:
+        logger.info(f'Latest tag in branch "{branch_name}": {latest_tag}')
+        if not re.match(r'^\d+\.\d+\.\d+$', latest_tag):
+            logger.info('Latest tag is not in format X.Y.Z.')
+        else:
+            suggested_tag = bump_minor_version(latest_tag)
+            logger.info(f'Suggested new version: {suggested_tag}')
+
+    while True:
+        user_input = input(
+            f'Input tag to release (press enter to use the suggested tag '
+            f"{suggested_tag or 'required'}), or 'c' to cancel: "
+        ).strip()
+
+        if user_input.lower() == 'c':
+            logger.warning('Release canceled.')
+            exit(0)
+
+        new_tag = user_input if user_input else suggested_tag
+
+        if not new_tag:
+            logger.error('Error: No tag specified and no suggestion available')
+            continue
+
+        if not re.match(r'^\d+\.\d+\.\d+$', new_tag):
+            logger.error('Error: Tag must be in format X.Y.Z')
+            continue
+
+        release_page = f'https://github.com/{owner}/{repo.name}/releases'
+        logger.warning(f'Check out the releases page: {release_page} before confirming!')
+
+        confirm = (
+            input(f"Confirm creating tag '{new_tag}' on branch '{branch_name}'? [y/N]: ")
+            .strip()
+            .lower()
+        )
+
+        if confirm == 'y':
+            break
+
+        logger.info("Let's try again...")
+
+    return new_tag
+
+
+def create_draft_release(
+    repo: github.Repository.Repository, tag: str, branch: str
+) -> github.GitRelease.GitRelease | None:
+    """Create a draft release with auto-generated notes."""
+    max_retries = 3
+    for attempt in range(1, max_retries + 1):
```

Comment:
Why is this needed? We don't have the retries everywhere else - is creating a release unreliable?

Reply:
I added this because in my network environment, I met failures here. I added retries elsewhere, making the script much more complicated, so I reverted most changes and kept only this one. But I agree it should be consistent, so I will remove this one too.